### PR TITLE
Added lxd_network module to manage LXD network profiles.

### PR DESCRIPTION
**Pull request by sirtaj**
##### SUMMARY

Adds new module lxd_network to manage network profile configuration on an LXD host. Code is based on existing cloud/lxd_profile module.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

cloud/lxd_network

##### ANSIBLE VERSION

```
ansible 2.6.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/sirtaj/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

### CODE REVIEW

File: lib/ansible/modules/cloud/lxd/lxd_network.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: lxd_network
+short_description: Manage LXD network
+version_added: 2.7
+description:
+  - Management of LXD networks
+author: "Sirtaj Singh Kang (@sirtaj) and Hiroaki Nakamura (@hnakamur)"
```
Comment (resmo) [inline lib/ansible/modules/cloud/lxd/lxd_network.py]: not a blocker but create a list for the authors (keep the key "author" as is)

~~~
- "Sirtaj Singh Kang (@sirtaj) 
- "Hiroaki Nakamura (@hnakamur)"
~~~

File: lib/ansible/modules/cloud/lxd/lxd_network.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+          - 'The config for the container (e.g. {"limits.memory": "4GB"}).
+            See U(https://github.com/lxc/lxd/blob/master/doc/rest-api.md#patch-3)'
+          - If the network already exists and its "config" value in metadata
+            obtained from
+            GET /1.0/networks/<name>
+            U(https://github.com/lxc/lxd/blob/master/doc/rest-api.md#get-19)
+            are different, they this module tries to apply the configurations.
+          - Not all config values are supported to apply the existing network.
+            Maybe you need to delete and recreate a network.
+        required: false
```
Comment (resmo) [inline lib/ansible/modules/cloud/lxd/lxd_network.py]: not a blocker: `required: false` is the default, can be omitted

**Comment by sirtaj** [comment]
I've resolved the issues left by @resmo, so I think this is ready to be merged.

**Comment by cedef** [comment]
Any news on this ?
Is LXD that little used over the world ? :S

**Comment by Nani-o** [comment]
I opened a PR 2 years ago with an lxd_network module, I use it alongside a drafted version of an lxd_storage module since then. I dont really care what PR will be merged but I would love those lxd modules to be merged.

**Comment by sirtaj** [comment]
I don't have a huge amount of faith in either PR being merged in a hurry. I'm not particularly attached to this one now since I'm not using LXD at the moment, which dulls the annoyance a bit.

**Pull request closed without merging**